File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/admin/indices/create/Alias.java
package fr.myprysm.vertx.elasticsearch.action.admin.indices.create;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class Alias {

    private String name;
    private String filter;
    private String indexRouting;
    private String searchRouting;

    /**
     * Build a new <code>Alias</code> from a <code>JsonObject</code>.
     *
     * @param json the <code>JsonObject</code>
     */
    public Alias(JsonObject json) {
        AliasConverter.fromJson(json, this);
    }

    /**
     * Transforms the <code>Alias</code> into a <code>JsonObject</code>.
     *
     * @return the <code>JsonObject</code>
     */
    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        AliasConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/admin/indices/create/Alias.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.admin.indices.create

import fr.myprysm.vertx.elasticsearch.action.admin.indices.create.Alias

fun Alias(
        filter: String? = null,
        indexRouting: String? = null,
        name: String? = null,
        searchRouting: String? = null): Alias = fr.myprysm.vertx.elasticsearch.action.admin.indices.create.Alias().apply {

    if (filter != null) {
        this.setFilter(filter)
    }
    if (indexRouting != null) {
        this.setIndexRouting(indexRouting)
    }
    if (name != null) {
        this.setName(name)
    }
    if (searchRouting != null) {
        this.setSearchRouting(searchRouting)
    }
}

